Description: If the web interface is activated by the CUPS configuration but
 the files for it (directories /usr/lib/cups/cgi-bin/ and
 /usr/share/cups/doc-root/) are not installed or unreadable, de-activate the
 web interface to enable auto-shutdown in on-demand mode.
Author: Till Kampetter <till.kampetter@gmail.com>
Last-Update: 2015-02-10

--- a/scheduler/conf.c
+++ b/scheduler/conf.c
@@ -3518,6 +3518,17 @@
                      sizeof(cupsd_vars) / sizeof(cupsd_vars[0]), cupsd_vars);
   }
 
+  /* Check if the files for the web interface are present, if not, set
+     the WebInterface variable to 0 to behave as there was no web interface */
+  struct stat filestats;
+  snprintf(temp, sizeof(temp), "%s/cgi-bin", ServerBin);
+  if (WebInterface &&
+      ((lstat(DocumentRoot, &filestats) || !S_ISDIR(filestats.st_mode) ||
+	S_ISLNK(filestats.st_mode) || !(filestats.st_mode & S_IROTH)) &&
+       (lstat(temp, &filestats) || !S_ISDIR(filestats.st_mode) ||
+	S_ISLNK(filestats.st_mode) || !(filestats.st_mode & S_IROTH))))
+      WebInterface = 0;
+
   return (1);
 }
 
